src/claimtrie.cpp

```diff
@@ -118,47 +173,82 @@ typename Container::value_type* getQueue(CDBWrapper& db, uint8_t dbkey, const Ke
     return nullptr;
 }
 
-uint256 computeHash(const std::vector<uint8_t>& vec)
+template <typename T>
+inline constexpr bool supportedType()
 {
-    return Hash(vec.begin(), vec.end());
+    static_assert(std::is_same<T, CClaimValue>::value || std::is_same<T, CSupportValue>::value, "T is unsupported type");
+    return true;
 }
 
-typename claimQueueType::value_type* CClaimTrieCacheBase::getQueueCacheRow(int nHeight, bool createIfNotExists)
+template <>
+std::pair<const int, std::vector<queueEntryType<CClaimValue>>>* CClaimTrieCacheBase::getQueueCacheRow(int nHeight, bool createIfNotExists)
 {
     return getQueue(*(base->db), CLAIM_QUEUE_ROW, nHeight, claimQueueCache, createIfNotExists);
 }
 
-typename queueNameType::value_type* CClaimTrieCacheBase::getQueueCacheNameRow(const std::string& name, bool createIfNotExists)
+template <>
+std::pair<const int, std::vector<queueEntryType<CSupportValue>>>* CClaimTrieCacheBase::getQueueCacheRow(int nHeight, bool createIfNotExists)
```

Comment:
how does this method name imply support?

Reply:
```
std::pair<const int, std::vector<queueEntryType<CSupportValue>>>*
                                                -------------
````

Reply:
Differentiating methods based on return value is not a good practice. Most computer languages don't even allow this; they don't include the return value type in the method footprint.

Reply:
It's perfectly valid in C++ and it's one of must have features to meta programming.